forc-pkg/src/lock.rs

```diff
@@ -34,7 +34,8 @@ pub struct PkgLock {
     version: Option<semver::Version>,
     // Short-hand string describing where this package is sourced from.
     source: String,
-    dependencies: Vec<PkgDepLine>,
+    dependencies: Option<Vec<PkgDepLine>>,
+    contract_dependencies: Option<Vec<PkgDepLine>>,
```

Comment:
out of curiosity, why are dependencies turned into an `Option` here?

Reply:
This was done so that, in the case that the package has no dependencies or contract dependencies, the `dependencies` or `contract_dependencies` field is omitted from the lock file entirely rather than shown with an empty list.

Reply:
Hmm so the change to `Option` isn't entirely related introducing contract dependencies? Was wondering if not having to specify dependencies was an intentional part of the change, or if it was just convenient to update this along the way.

Reply:
Good question! The reason why I gave the pass for this change in this PR is that a decision needed to be made for the new `contract-dependencies` field. Without the `Option`, every test lock file in the Sway repo would require a new empty `contract-dependencies` entry, so Kaya added the `Option`. Then we figured the behaviour should be consistent for both of the dependencies fields, so the decision was made to have `dependencies` use an `Option` too.

Reply:
Ah, got it, that makes sense, thanks :+1: